server/schema.ts

```diff
@@ -0,0 +1,84 @@
+import { makeExecutableSchema } from "graphql-tools";
+import * as fetch from "isomorphic-unfetch";
+
+// Types
+const typeDefs = `
+  type Query {
+    categories: [Category]
+    exercises(category: Int!) : [Exercise]
+    exercise(id: Int!) : Exercise
+    exerciseimage(id: Int!) : [ExerciseImage]
+  }
+
+  type Category {
+    id: Int!
+    name: String!
+  }
+
+  type Exercise {
+    id: Int!
+    license_author: String
+    status: String
+    description: String
+    name: String
+    name_original: String
+    creation_data: String
+    uuid: String
+    license: Int
+    category: Int
+    language: Int
+    muscles: [Int]
+    muscles_secondary: [Int]
+    equipment: [Int]
+    image: [ExerciseImage]
+  }
+
+  type ExerciseImage {
+    id: Int!
+    license_author: String
+    status: String
+    image: String
+    is_main: Boolean
+    license: Int
+    exercise: Int!
+  }
+`;
+
+// Resolvers
+const resolvers = {
+  Query: {
+    categories: async () => {
+      const r = await fetch("https://wger.de/api/v2/exercisecategory");
+      const data = await r.json();
+
+      return data.results;
+    },
+    exercise: async (_, { id }) => {
+      const info = fetch(`https://wger.de/api/v2/exercise/${id}`);
+      const image = fetch(
+        `https://wger.de/api/v2/exerciseimage?exercise=${id}`,
+      );
+      const [rInfo, rImage] = [await info, await image];
+      const [dataInfo, dataImage] = [await rInfo.json(), await rImage.json()];
+
+      return {
+        ...dataInfo,
+        image: dataImage.results,
+      };

```

Comment:
Here you can also use Promise all to not block per await:
```typescript
      const info = getJSON(`https://wger.de/api/v2/exercise/${id}`);
      const image = getJSON(
        `https://wger.de/api/v2/exerciseimage?exercise=${id}`,
      );

      const [dataInfo, dataImage] = await Promise.all([info, image]);

      return {
        ...dataInfo,
        image: dataImage.results,
      };
```

Reply:
According to [this SO answer](https://stackoverflow.com/a/45772148), it should work fine. It seems to work fine in [this JS Bin](http://jsbin.com/cehujotemu/edit?js,console) (You have to open your browser console to see the logged time). Or did I miss something?

Reply:
Didn't know about that one, tested it a bit myself and it seems to act the same way as `Promise.all`. So I guess either is fine to use!